kartothek/io_components/metapartition.py

```diff
@@ -1549,6 +1549,32 @@ def delete_from_store(self, dataset_uuid, store):
             store.delete(file_key)
         return self.copy(files={}, data={}, metadata={})
 
+    def get_parquet_metadata(self, store, table_name):
+        """
+        Retrieve the parquet metadata, especially relevant for dataset statistics.
+
+        # TODO docstrings
+        """
+        if not isinstance(table_name, str):
+            raise TypeError("Expecting a string for parameter `table_name`.")
+
+        if self.files:
+            fd = store.open(self.files[table_name])
+            pq_metadata = pa.parquet.ParquetFile(fd).metadata
+            stats = {
+                "number_rows": pq_metadata.num_rows,
+                "number_row_groups": pq_metadata.num_row_groups,
+                "serialized_size": pq_metadata.serialized_size,
+            }
+            for rg_nb in range(stats["number_row_groups"]):
+                stats[f"row_group_{rg_nb}_byte_size"] = pq_metadata.row_group(
```

Comment:
This will probably not work. It is highly likely that the files in a dataset have different numbers of rowgroups. This would then return you a different schema for every partition. If you want to expose the RG information you might need to blow up the data, e.g.

| partition label | number of rows | number of row groups | row group ID |  row group byte size | 
| - |  - | - | - |  - | 
| "label" |  10000 | 2 | 1 |  5 | 
| "label" |  10000 | 2 | 2 |  20 | 

wherre the partition-level rows are duplicated

The primary key of this table is then `("partition label", "row group ID")`

Reply:
Makes sense. I'll keep the row group information and blow up the data.